content/articles/online-machine-learning-with-river-python/index.md

```diff
@@ -0,0 +1,446 @@
+
+In online machine learning, the model uses real-time data fo training and makes a sigle observation at a time.
+We then update our model as each new data arrives, this increases the model accuracy over time. The input data is continuously used to improve the current model's knowledge and as we further train the model becomes more adaptive and robust.
+
+### River Python installation
+
+Since we are using Google Colab in this tutorial, we install using this command.
+
+```python
+!pip install river
+```
+
+#### Import River
+
+To use River we import it into our machine.
+
+```python
+import river
+```
+
+### Checking methods and attributes
+
+The River Python has various methods and attributes for online machine learning. To list all the available methods use the following command.
+
+```python
+dir(river)
+```
+
+List of River methods and attributes
+
+```bash
+['__all__',
+ '__builtins__',
+ '__cached__',
+ '__doc__',
+ '__file__',
+ '__loader__',
+ '__name__',
+ '__package__',
+ '__path__',
+ '__spec__',
+ '__version__',
+ 'anomaly',
+ 'base',
+ 'cluster',
+ 'compat',
+ 'compose',
+ 'datasets',
+ 'drift',
+ 'dummy',
+ 'ensemble',
+ 'evaluate',
+ 'expert',
+ 'facto',
+ 'feature_extraction',
+ 'feature_selection',
+ 'imblearn',
+ 'linear_model',
+ 'meta',
+ 'metrics',
+ 'multiclass',
+ 'multioutput',
+ 'naive_bayes',
+ 'neighbors',
+ 'neural_net',
+ 'optim',
+ 'preprocessing',
+ 'proba',
+ 'reco',
+ 'stats',
+ 'stream',
+ 'synth',
+ 'time_series',
+ 'tree',
+ 'utils']
+```
+
+Some of the methods that we will be using are as follows.
+
+- `naive_bayes`
+  This is the algorithm that we will be using to build our text classification model.
+
+- `preprocessing`
+  Will be used in the processing of our dataset used to train our model.
+
+- `metrics`
+  Calculate the accuracy score of our model.
+
+- `stream`
+  Will be used to simulate our dataset to be streaming data.
+
+- `anomaly`
+  Will be used to detect errors and anomalies in our model.
+
+- `compose`
+  Used to build a pipeline to automate machine learning workflows.
+
+These methods are very helpful in this tutorial.
+
+### Loading machine learning packages
+
+```python
+import MultinomialNB from river.naive_bayes
+import BagOfWords,TFIDF from river.feature_extraction
+```
+
+We have imported the following packages:
+
+#### MultinomialNB
+
+This is a Naive Bayes method that is used for text classification. This contains the algorithm that is useful in the building of our model.
+
+For a detailed understanding about MultinomialNB [click here](https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.MultinomialNB.htmll)
+
+#### BagOfWords
+
+Is used to extract various features in our dataset. Features are the independent variables that are used as input for our model.
+
+It also converts our text inputs into vectors that are more machine-readable. BagOfWords work the same as `CountVectorizer` in offline machine learning.
```

Comment:
Please explain in a few words how `BagOfWords` work the same as `CountVectorizer` before you refer your reader to the external link.